src/test/regress/expected/failure_savepoints_0.out

```diff
@@ -0,0 +1,348 @@
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SET citus.shard_count = 2;
+SET citus.shard_replication_factor = 1; -- one shard per worker
+SET citus.next_shard_id TO 100950;
+ALTER SEQUENCE pg_catalog.pg_dist_placement_placementid_seq RESTART 150;
+CREATE TABLE artists (
+    id bigint NOT NULL,
+    name text NOT NULL
+);
+SELECT create_distributed_table('artists', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- add some data
+INSERT INTO artists VALUES (1, 'Pablo Picasso');
+INSERT INTO artists VALUES (2, 'Vincent van Gogh');
+INSERT INTO artists VALUES (3, 'Claude Monet');
+INSERT INTO artists VALUES (4, 'William Kurelek');
+-- simply fail at SAVEPOINT
+SELECT citus.mitmproxy('conn.onQuery(query="^SAVEPOINT").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+BEGIN;
+INSERT INTO artists VALUES (5, 'Asher Lev');
+SAVEPOINT s1;
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:9060
+WARNING:  connection error: localhost:9060
+DETAIL:  connection not open
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:9060
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:9060
+ERROR:  connection not open
+CONTEXT:  while executing command on localhost:9060
+DELETE FROM artists WHERE id=4;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+RELEASE SAVEPOINT s1;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+COMMIT;
+SELECT * FROM artists WHERE id IN (4, 5);
+ id |      name       
+----+-----------------
+  4 | William Kurelek
+(1 row)
+
+-- fail at RELEASE
+SELECT citus.mitmproxy('conn.onQuery(query="^RELEASE").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+BEGIN;
+UPDATE artists SET name='a';
+SAVEPOINT s1;
+DELETE FROM artists WHERE id=4;
+RELEASE SAVEPOINT s1;
+WARNING:  AbortSubTransaction while in COMMIT state
```

Comment:
This looks suspicious. Could it be a bug in our savepoint implementation? (We have one more of this.)

It looks like we're first sending `RELEASE SAVEPOINT savepoint_2`. Later, after a failure, we try to rollback to `savepoint_2`

```SQL
LOG:  issuing RELEASE SAVEPOINT savepoint_2
DETAIL:  on server localhost:9700
LOG:  issuing RELEASE SAVEPOINT savepoint_2
DETAIL:  on server localhost:9701
LOG:  issuing RELEASE SAVEPOINT savepoint_2
DETAIL:  on server localhost:9702
LOG:  issuing RELEASE SAVEPOINT savepoint_2
DETAIL:  on server localhost:9700
LOG:  issuing RELEASE SAVEPOINT savepoint_2
DETAIL:  on server localhost:9700
WARNING:  AbortSubTransaction while in COMMIT state
LOG:  issuing ROLLBACK TO SAVEPOINT savepoint_2
DETAIL:  on server localhost:9700
WARNING:  could not issue cancel request
DETAIL:  Client error: PQcancel() -- connect() failed: Connection refused

WARNING:  connection not open
```

Reply:
I'm OK (and prefer) to not fix it right now. But we should open an issue if you think so as well.

Reply:
COMMIT state means RELEASE started, but it didn't end. It is more like "COMMITTING" not "COMMITTED". So it doesn't look an issue to me.

Reply:
Noting what we've talked in person, the warning is coming from the coordinator, so look inevitable.